src/Home/CreatePlayerButton.js

```diff
@@ -5,15 +5,31 @@ import CreateRoomButton from "./CreateRoomButton";
 
 const CreatePlayerButton = (props) => {
   const [usernameInput, setUsernameInput] = useState();
+  const [usernameTaken, setUsernameTaken] = useState(false);
   const { player, setPlayer, setPage, setRoomCode, setIsHost } = props;
   const onChange = (e) => {
     setUsernameInput(e.target.value);
   };
+
+  const fetchPlayerNames = async () => {
+    const { data } = await supabase.from("Players").select("name");
+    const registeredPlayers = data;
+    return registeredPlayers;
+  };
   const handleClick = async () => {
-    setPlayer(usernameInput);
-    const { error, status } = await supabase
-      .from("Players")
-      .insert([{ name: usernameInput }]);
+    const takenPlayerNames = await fetchPlayerNames();
+    const takenName = takenPlayerNames.find(
+      (playerNames) => playerNames.name === usernameInput
+    );
+    console.log(takenName);
+    if (takenName === undefined) {
+      setPlayer(usernameInput);
+      const { error, status } = await supabase
+        .from("Players")
+        .insert([{ name: usernameInput }]);
+    } else {
+      setUsernameTaken(true);
+    }
```

Comment:
If you dont want to allow for people to have the same username, a better way might be to enforce uniqueness on the column in supabase/the DB. So instead of getting the entire list of players from supa, and then running a .find (O(n)) and then making a post request, we could just do the post request and see if it comes back with a conflict error.

I dont think we need to enforce unique names though.

Reply:
You are right, it is not entirely necessary to implement unique player names. It was just implemented to ensure no two players in a room have the same name, which would cause issues with updating their hit points and other relevant information.